src/controller/orderController.js

```diff
@@ -1,47 +1,53 @@
-const CartModel = require("../models/cart");
+const cartModel = require("../models/cart");
 const productModel = require("../models/product");
 const orderModel = require("../models/order");
 
 // Create a Order
-exports.cheakOut = async (req, res) => {
+exports.checkOut = async (req, res) => {
   try {
     // Find the Cart products Of the specific user
-    const { cartProducts } = await CartModel.findOne(
-      {
-        userId: req.params.userId,
-      },
-      { products: 1, _id: 0 }
-    ).lean();
+    const cartProducts = await cartModel
+      .findOne(
+        {
+          userId: req.params.userId,
+        },
+        { products: 1, _id: 0 }
+      )
+      .lean();
 
     // Maping the the product Id from cart Products
-    const allProductId = cartProducts.map((product) => product.productId);
+    const allProductId = cartProducts.products.map(
+      (product) => product.productId
+    );
 
     // Find the product details from the product Collection which id is in allProductId
     const allProduct = await productModel.find({ _id: { $in: allProductId } });
 
     // Finding the Total Amount of All Product in cart
     let totalBill = 0;
     allProduct.forEach((productCollectionProducts) => {
-      cartProducts.forEach((cartProductData) => {
-        if (productCollectionProducts._id == cartProductData.productId) {
-          totalBill +=
-            productCollectionProducts.price * cartProductData.quantity;
+      cartProducts.products.forEach((cartProductData) => {
+        const { _id, price } = productCollectionProducts;
+        const { productId, quantity } = cartProductData;
+
```

Comment:
you can destructure above also
`cartProducts.products.forEach(({ productId, quantity ) => {`